### PR TITLE
Draft: data pipeline scripts for pose segmentation and batching

### PR DESCRIPTION
**Pull request by ImYangYun**
About `signwriting_animation/data/`:

I'm working on a data preparation pipeline for my thesis project, which aims to generate skeletal pose animations from SignWriting input.
- `collect_pose.py`: extracts and saves raw skeletal pose data from JSON or NPZ files into structured NumPy arrays.
- `segment_poses.py`: applies segmentation based on a CSV file containing start and end frame indices. Segments shorter than a defined minimum length are filtered out. Each segmentation file corresponds to one pose file.
- `data_loader.py`: defines a PyTorch `Dataset` that returns pose clips of fixed window size. The `collate_fn` applies padding and masking to prepare batched input for Transformer-style models.

I’ve tested the pipeline, including batching. The output is a structured dictionary containing `pose`, `mask`, and `metadata`, and the loader works correctly with variable-length segments. 

_Note: most earlier commits(file renames, edits, etc.) are minor and can be ignored._

### CODE REVIEW

File: signwriting_animation/data/collect_pose.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    print(f"\n Done! Saved metadata to {output_csv}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Collect pose metadata from .pose files")
+    parser.add_argument("--pose_dir", required=True, help="Directory containing .pose files")
+    parser.add_argument("--output", required=True, help="Output CSV path")
+
+    args = parser.parse_args()
+    collect_pose_metadata(args.pose_dir, args.output)
```
Comment (AmitMY) [inline signwriting_animation/data/collect_pose.py:33]: why are you collecting the metadata yourself?
This is the full CSV https://github.com/sign/data/blob/main/signwriting-transcription/data.csv

File: signwriting_animation/data/segment_poses.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# get_data.py
+import os
+import pandas as pd
+import argparse
+
+def create_fixed_segments(data_csv, output_csv, window_size=60):
```
Comment (AmitMY) [inline signwriting_animation/data/segment_poses.py:6]: there's no need to create segments and save them on disk.
You can do this on the fly - load the pose file, then randomally choose a frame to cut until for context, and cut from for the generated data. using fixed windows might not be beneficial here

File: signwriting_animation/data/data_loader.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    def __len__(self):
+        return len(self.samples)
+
+    def __getitem__(self, idx):
+        item = self.samples[idx]
+        with open(item["path"], "rb") as f:
+            pose = Pose.read(f.read())
+
+        data = pose.body.data.data
```
Comment (AmitMY) [inline signwriting_animation/data/data_loader.py:50]: you can do something like
```py
past_pose = pose.body[start:end]
future_pose = pose.body[end:end+self.future_frames]
```
instead of multiple slices - as in, you can slice directly the pose object

File: signwriting_animation/data/data_loader.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        clip_mask = mask[item["start"]:item["end"]]
+
+        input_pose = clip[:self.past_frames]
+        target_pose = clip[self.past_frames:]
+        input_mask = clip_mask[:self.past_frames]
+
+        result = {
+            "input_pose": torch.tensor(input_pose, dtype=torch.float32),
+            "target_pose": torch.tensor(target_pose, dtype=torch.float32),
+            "mask": torch.tensor(input_mask, dtype=torch.bool),
```
Comment (AmitMY) [inline signwriting_animation/data/data_loader.py:63]: i assume there will also need to be a `input_mask` and a `target_mask`?

File: signwriting_animation/data/data_loader.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        }
+
+        if self.with_metadata:
+            result["metadata"] = {
+                "total_frames": item["end"] - item["start"]
+            }
+
+        return result
+
+def collate_pose_fn(batch):
```
Comment (AmitMY) [inline signwriting_animation/data/data_loader.py:74]: you could also replace this with our zero pad collator - https://github.com/sign-language-processing/pose/blob/4c9103d437fe97ebf1251bfc0c1c5af606743b89/src/python/pose_format/torch/masked/collator.py#L47

**Review by AmitMY — CHANGES_REQUESTED** [review]
